### PR TITLE
docs(iroh-net): deny missing docs and broken links in the crate, except the `derp` mod

### PR DESCRIPTION
**Pull request by divagant-martian**
the individual `deny` should be moved to the top level once the `derp` is documented
part of #1156

### CODE REVIEW

File: iroh-net/src/hp/stun.rs [hunk 1]
```diff
@@ -11,23 +13,30 @@ pub use stun_rs::{
 
 use crate::net::ip::to_canonical;
 
+/// Errors that can occurr when handling a STUN packet.
 #[derive(Debug, thiserror::Error)]
 pub enum Error {
+    /// The STUN message could not be parsed or is otherwise invalid.
     #[error("invalid message")]
     InvalidMessage,
+    /// STUN request is not a binding request when it should.
```
Comment (ramfox) [inline iroh-net/src/hp/stun.rs:22]:
```suggestion
    /// STUN request is not a binding request when it should be.
```

File: iroh-net/src/hp/stun.rs [hunk 2]
```diff
@@ -11,23 +13,30 @@ pub use stun_rs::{
...
     #[error("not binding")]
     NotBinding,
+    /// STUN packet is not a response when it should.
```
Comment (ramfox) [inline iroh-net/src/hp/stun.rs:25]:
```suggestion
    /// STUN packet is not a response when it should be.
```

**Review by ramfox — APPROVED** [review]
two small grammar nits, but otherwise looks good!

**Comment by divagant-martian** [comment]
I commited your suggestions @ramfox and the approval is dismissed :roll_eyes:

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**